src/types.rs

```diff
@@ -1388,3 +1388,61 @@ define_bounded_integers! {
     BoundedI64 i64 signed -> i64,
     BoundedIsize isize signed -> isize,
 }
+
+//Indexing operations on [T; N], Vec<T> and VecDeque<T> for BoundedUsize
+#[cfg(all(feature = "std", feature = "types"))]
+impl<const MIN: usize, const MAX: usize, const N: usize, T> std::ops::Index<BoundedUsize<MIN, MAX>>
+    for [T; N]
+{
+    type Output = T;
+
+    #[inline]
+    fn index(&self, index: BoundedUsize<MIN, MAX>) -> &Self::Output {
+        &self[index.get()]
+    }
+}
+
+#[cfg(all(feature = "std", feature = "types"))]
+impl<const MIN: usize, const MAX: usize, T> std::ops::Index<BoundedUsize<MIN, MAX>>
```

Comment:
One more thing: can you add implementations for `IndexMut` as well? I don’t think it should be much more code.

Reply:
yeah sure! It's a bit late rn so I might finish up tomorrow but these are good ideas